core/sds-aspect-model-java-generator/src/main/resources/java-static-class-body-lib.vm

```diff
@@ -75,6 +75,19 @@ static {
        );
     }
 
+    @Override
+    public List<StaticProperty<?>> getAllProperties() {
+        #if( $element.getExtends().isPresent() )
+            #set( $extendedElement = $element.getExtends().get() )
+            $codeGenerationConfig.getImportTracker().importExplicit( "${codeGenerationConfig.getPackageName()}.Meta${extendedElement.getName()}" )
+            List<StaticProperty<?>> properties = getProperties();
+            properties.addAll(Meta${extendedElement.getName()}.INSTANCE.getProperties());
```

Comment:
I think this method does not handle multiple steps of inheritance, i.e. grand-parent's (and higher) properties. Shouldn't this line have `INSTANCE.getAllProperties()`?

Additionally, I think there is no test for this case.

Reply:
Corrected.

There is a test where we validate against the method of the generated files, including one for a grand-parent.